### PR TITLE
Fix InterruptedException when KafkaCollectorWorker stops

### PR DESCRIPTION
**Pull request by gquintana**
Fixes https://github.com/openzipkin/zipkin/issues/2448

### CODE REVIEW

**Comment by jeqo** [comment]
Thanks @gquintana !

LGTM

**Review by jeqo — APPROVED** [review]

**Comment by codefromthecrypt** [comment]
ps I checked Gerald already has an ASF iCLA http://people.apache.org/unlistedclas.html

While we haven't yet moved this repo, it will soon.

File: zipkin-collector/kafka/src/main/java/zipkin2/collector/kafka/KafkaCollectorWorker.java [hunk 1]
```diff
@@ -78,8 +82,8 @@ public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
           }
         });
       LOG.info("Kafka consumer starting polling loop.");
-      while (true) {
-        final ConsumerRecords<byte[], byte[]> consumerRecords = kafkaConsumer.poll(1000);
+      while (running.get()) {
+        final ConsumerRecords<byte[], byte[]> consumerRecords = kafkaConsumer.poll(Duration.of(1000, ChronoUnit.MILLIS));
```
Comment (codefromthecrypt) [inline zipkin-collector/kafka/src/main/java/zipkin2/collector/kafka/KafkaCollectorWorker.java:86]: this change is fine as our source level is 1.8 now (for collectors)

File: zipkin-collector/kafka/src/main/java/zipkin2/collector/kafka/KafkaCollectorWorker.java [hunk 2]
```diff
@@ -103,18 +107,21 @@ public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
...
-      LOG.info("Closing Kafka consumer...");
-      LOG.info("Kafka consumer closed.");
+      LOG.info("Kafka consumer polling loop stopped. Kafka consumer closed.");
     }
   }
 
+  /**
+   * Stop the polling loop
+   */
+  public void stop() {
```
Comment (codefromthecrypt) [inline zipkin-collector/kafka/src/main/java/zipkin2/collector/kafka/KafkaCollectorWorker.java:121]: nit this needn't be public, but I'm not concerned about it enough to ask for a revision.

**Review by codefromthecrypt — APPROVED** [review]
Thanks for the help!

**Pull request merged**